### PR TITLE
Stabilize CI tests

### PR DESCRIPTION
**Pull request by chrismayer**
This PR fixes the `package-lock.json,` which was messed up since merging the PR #335, see #342 for details. The file is re-created with Node.js v16 and npm in v8. Since Node,js [v14 has reached its end-of-life](https://github.com/nodejs/release#release-schedule) this PR also updates the engines within the `package.json`. 

This also changes the CI test pipeline to use `npm ci` instead of `npm i` to have a reliable state in the tests. Kudos to @sronveaux , who proposed this idea in #343.

### CODE REVIEW

**Comment by sronveaux** [comment]
Hi @chrismayer,

Thanks for the PR and kind word here, however, by running through `Wegue` repository I came across #294 opened by @JakobMiksch last year which was talking about the same issue... so it looks `npm ci` was already a concern which was put aside until `Wegue V2` which means... now!

So kudos are not for me on this one ;-)

**Review by sronveaux — APPROVED** [review]
Except for potential detail about `npm` version where a decision should be taken, seems perfect on my side !

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "vue-template-compiler": "2.7.14"
   },
   "engines": {
-    "node": ">= 14.15.0",
-    "npm": ">= 6.14.8"
+    "node": ">= 16.19.0",
+    "npm": ">= 8.19.0"
```
[the remark this answers is not in the document]
Comment (chrismayer) [reply package.json]: Totally agree, I set the npm version to `8.19.3` with f3d735d.

**Comment by chrismayer** [comment]
Thanks for your review @sronveaux. I am going to merge now.

**Comment by sronveaux** [comment]
Hi @chrismayer ,

Just thought about something here, I saw that the `Dockerfile` is also using `npm install` where it could use `npm ci` too...

This depends whether the docker implementation is aimed at development or at deployment. In the first case, it sould stay with `npm install`. In the second case, it should be changed to `npm ci`.

Can you give the underlying idea of the docker implementation to see whether it should be adapted or not ?  I can make the adaptation if needed, I just tested it and it works flawlessly...

**Pull request merged**